src/register/pmpaddrx.rs

```diff
@@ -27,30 +27,51 @@ impl PmpAddr {
     }
 
     #[inline]
-    pub fn encode(&mut self, mode: Mode, addr: Addr, size: Option<NonZeroSize>) {
+    pub fn encode(&mut self, mode: Mode, addr: Addr, size: Option<NonZeroSize>) -> Result<(), ()> {
         self.bits = match mode {
             Mode::OFF => 0,
-            Mode::TOR => (addr >> 2).try_into().unwrap(),
-            Mode::NA4 => (addr >> 2).try_into().unwrap(),
-            Mode::NAPOT => Self::encode_napot(addr, size.unwrap().into()),
-        }
+            Mode::TOR => {
+                let addr_small: usize = (addr >> 2) as usize;
+                if (addr_small as u64) << 2 != addr {
+                    return Err(());
+                }
```

Comment:
Some standard library crate should provide `align_up`/`align_down` functionality. It would be so nice to use that in such scenarios to make code more readable. I found implementations of these in some x86_64 crate, but not in any generic library crate.

Reply:
It may be worth making our own with it or adding it to this crate somewhere.  I think this has come up in a few different PRs now.  
This check ensures both the low 2 bits and high 30 bits were not truncated though, so some additional logic would still be needed.

Reply:
Ack.